### DISCUSSION TITLE
uploadListingImage - building the multipart/form-data

### DISCUSSION THREAD

**Discussion by cmunden** [etsy/open-api#1023] [Q&A]
When using the uploadListingImage call, how should the extraneous parameters (rank, overwrite, is_watermarked, alt_text) of the request be handled?  Are they URL encoded and added to the baseURL of the call or are they added as separate parts to the multipart body and specified using a content-disposition header for each?

If someone could provide an example of what the resulting multipart content looks like for a sample request, it would be very helpful in knowing how to handle or implement the call.

**Comment by adiljaved76** [comment]
RestClient restClient = new RestClient();
                RestRequest request;

                request = new RestRequest(url, EtsyUtilities.EtsyUploadListingImageEP.Method);

                request.AddHeader("x-api-key", _etsyService.EtsySetting.EtsyApiKey);
                request.AddHeader("Authorization", "Bearer " + _etsyService.EtsySetting.EtsyAccessToken);

                if (!String.IsNullOrWhiteSpace(finalPath))
                    request.AddFile("image", finalPath);

                if (image.ImageID > 0)
                    request.AddParameter("listing_image_id", image.ImageID);

                request.AddParameter("rank", image.Rank);
                request.AddParameter("overwrite", image.Overwrite);
                request.AddParameter("is_watermarked", image.IsWaterMarked);
                request.AddParameter("alt_text", image.AltText);

                RestResponse response = restClient.Execute(request);

                if (!response.IsSuccessStatusCode)
                {
                    throw new Exception(response.Content, response.ErrorException);
                }

**Reply by adiljaved76** [reply]
this is how I'm doing in C#
